src/internal/task/etcd_service.go

```diff
@@ -162,60 +168,57 @@ func (ed *etcdDoer) Do(ctx context.Context, inputChan chan *types.Any, cb Collec
 				}
 				return nil
 			})
-			return errors.EnsureStack(err)
-		})
-		defer func() {
-			if _, err := col.NewSTM(ctx, ed.etcdClient, func(stm col.STM) error {
-				if err := ed.taskCol.ReadWrite(stm).DeleteAllPrefix(prefix); err != nil {
-					return errors.EnsureStack(err)
-				}
-				return errors.EnsureStack(ed.claimCol.ReadWrite(stm).DeleteAllPrefix(prefix))
-			}); err != nil {
-				fmt.Printf("errored deleting tasks with the prefix %v: %v\n", prefix, err)
+			if err != nil && !errors.As(err, context.Canceled) {
```

Comment:
I don't think this is correct. If we finish laying out the work and the context gets canceled by the caller while waiting, we would return nil rather than context canceled.